libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Common/SalesforceSDKManager.m

```diff
@@ -846,6 +851,31 @@ - (void)biometricAuthenticationFlowDidComplete:(NSNotification *)notification {
     return [SFScreenLockManagerInternal shared];
 }
 
+#pragma mark - Native Login
+
+- (id <SFNativeLoginManager>)useNativeLoginWithConsumerKey:(nonnull NSString *)consumerKey
+                                               callbackUrl:(nonnull NSString *)callbackUrl
+                                              communityUrl:(nonnull NSString *)communityUrl
+                                 nativeLoginViewController:(nonnull UIViewController *)nativeLoginViewController
+                                                     scene:(nullable UIScene *)scene {
+    
+    [SFSDKAppFeatureMarkers registerAppFeature:kSFAppFeatureNativeLogin];
+    NSString *key = scene ? scene.session.persistentIdentifier : kSFDefaultNativeLoginViewControllerKey;
+    [_nativeLoginViewControllers setObject:nativeLoginViewController forKey:key];
+    nativeLogin = [[SFNativeLoginManagerInternal alloc] initWithClientId:consumerKey redirectUri:callbackUrl loginUrl:communityUrl scene:scene];
```

Comment:
it looks like if a couple of scenes were opened in a row that manager would get overwritten and only have the latest one? I wasn't thinking about that when we were talking about it before, wondering if login could also have a scene parameter so you don't have to store scene directly on the manager and this could only be initialized once? or could have a property so the SDK can create the view controllers as needed like `SnapshotViewControllerCreationAction` or `SFLoginViewControllerCreationBlock`

Reply:
I will think that over for unexpected side effects, but I did design it like this intentionally.  I'm betting that 90%+ of apps that use this will have one native login manager so we should hold one reference they can retrieve for convenience.  However, since these managers are **per community/org** there will be a more rare use case for multiple (prod + sandbox or perhaps publisher's public playground app).  Apps that need that can maintain references to managers themselves.  Although I haven't given much thought to that being a memory leak.  

Regardless I _think_ I need to store the scene with the manager because it links the result of the rest calls being made from that login VC to the auth session in the SDK that is necessary for correctly creating the user account.